spec/models/solidus_bolt/payment_source_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+
+RSpec.describe SolidusBolt::PaymentSource, type: :model do
+  let(:column_list) {
+    [
+      'id',
+      'payment_method_id',
+      'transaction_id',
+      'transaction_reference',
+      'transaction_type',
+      'processor',
+      'date',
+      'transaction_status',
+      'amount',
+      'currency',
+      'created_at',
+      'updated_at'
+    ]
+  }
+
+  describe 'Check Model Integrity' do
+    it 'has correct table name' do
+      expect(described_class.table_name).to eq 'solidus_bolt_payment_sources'
+    end
+
+    it 'has correct attributes' do
+      expect(described_class.new.attributes.keys.sort).to eq column_list.sort
+    end
+  end
+
+  describe 'transaction_reference' do
+    context 'when transaction_reference is not present' do
+      it 'has invalid object' do
+        object = described_class.new
+        expect(object.valid?).to eq(false)
+      end
+
+      it 'has an error' do
+        object = described_class.new
+        object.save
```

Comment:
the errors appear also after `.valid?` which is faster to process than `.save`

Reply:
used `.valid?` instead of `.save` now